resources/lib/apihelper.py

```diff
@@ -353,10 +353,13 @@ def get_upnext(self, info):
                   program=program, season=season, episode=current_ep_no)
         return None
 
-    def get_single_episode(self, whatson_id):
-        ''' Get single episode by whatsonId '''
+    def get_single_episode(self, ep_id, upnext=False):
+        ''' Get single episode by whatsonId or videoId '''
         video = None
-        api_data = self.get_episodes(whatson_id=whatson_id, variety='single')
+        if upnext:
+            api_data = self.get_episodes(video_id=ep_id, variety='single')
```

Comment:
I am not so happy with this interface. The upnext boolean is actually a switch to decide whether it is a video_id or a whatson_id. I much rather have a more explicit interface:
```python
def get_single_episode(self, video_id=None, whatson_id=None):
```

*Or possibly two entry points instead of a shared one.*

Reply:
Okay, changed.